pkgs/applications/science/electronics/vivado/default.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv
+, coreutils
+, fetchurl
+, patchelf
+, procps
+, makeWrapper
+, requireFile
+, ncurses5
+, zlib
+, libuuid
+, libSM
+, libICE
+, libX11
+, libXrender
+, libxcb
+, libXext
+, libXtst
+, libXi
+, glib
+, freetype
+, gtk2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vivado";
+  version = "2017.2";
+
+  buildInputs = [
+    patchelf
+    procps
+    ncurses5
+    makeWrapper
+    coreutils
+  ];
+
+  builder = ./builder.sh;
+  inherit ncurses5;
+
+  # requireFile prevents rehashing each time, which saves time during
+  # rebuilds.
+  src = requireFile rec {
+    name = "Xilinx_Vivado_SDK_2017.2_0616_1.tar.gz";
+    message = ''
+      This nix expression requires that ${name} is already part of the store.
+      Login to Xilinx, download from
+      https://www.xilinx.com/support/download.html,
+      rename the file to ${name}, and add it to the nix store with
+      "nix-prefetch-url file:///path/to/${name}".
+    '';
+    sha256 = "06pb4wjz76wlwhhzky9vkyi4aq6775k63c2kw3j9prqdipxqzf9j";
+  };
```

Comment:
Can we make the `requireFile` arguments (`name` and `sha256`) arguments with default values to the derivation (i.e. after line 21)? That'll make it a bit easier for people to upgrade on their own (with `vivado.override { name = "..."; sha256 = "..."; }` rather than making an overlay) after Xilinx releases new versions that haven't been hashed/submitted here yet.

Reply:
@9999years definitely. The only thing is I can't guarantee the builder will work for anything other than this version. For instance, the builder failed for 2019.

Reply:
> with `vivado.override { name = "..."; sha256 = "..."; }` rather than making an overlay

> For instance, the builder failed for 2019.

Than we should not add this and let people overwrite it with overwriteAttr